packages/x-data-grid/src/models/gridBaseSlots.ts

```diff
@@ -118,6 +118,7 @@ export type CheckboxProps = {
   onKeyDown?: React.KeyboardEventHandler;
   size?: 'small' | 'medium';
   density?: 'standard' | 'compact';
+  truncate?: boolean;
```

Comment:
Would it make sense to enable this behavior by default for all checkboxes? It would keep the base typings more tidy.

Reply:
Nice suggestion, have made it the default behaviour and removed the prop 👍